string subject
string target
string action
---
bool exists

